### PR TITLE
Add isParentOf, isChildOf, depthRelatedTo

### PR DESCRIPTION
**Pull request by PaperTurtle**
This merge request add three new methods mentioned in (#179) to the HasRecursiveRelationships Trait.

I've tried to create a test for the depthRelatedTo, but I cannot get it to work.

However, I have created a Laravel Project to test if the methods work accordingly. I've created an example Model: 
```php
class Category extends Model
{
    use HasFactory, HasRecursiveRelationships;

    protected $fillable = ['name', 'parent_id'];
}
```

and then tried the methods out: 
```php
Route::get('/setup-multi-level-categories', function () {
    $rootCategory = Category::create(['name' => 'Root Category']);

    $firstLevelCategory = Category::create([
        'name' => 'First Level Category',
        'parent_id' => $rootCategory->id
    ]);

    $secondLevelCategory = Category::create([
        'name' => 'Second Level Category',
        'parent_id' => $firstLevelCategory->id
    ]);

    $thirdLevelCategory = Category::create([
        'name' => 'Third Level Category',
        'parent_id' => $secondLevelCategory->id
    ]);

    return 'Multi-level categories setup completed.';
});

Route::get('/is-parent-of/{parentId}/{childId}', function ($parentId, $childId) {
    $parent = Category::findOrFail($parentId);
    $child = Category::findOrFail($childId);

    return response()->json([
        'isParentOf' => $parent->isParentOf($child)
    ]);
});

Route::get('/is-child-of/{childId}/{parentId}', function ($childId, $parentId) {
    $child = Category::findOrFail($childId);
    $parent = Category::findOrFail($parentId);

    return response()->json([
        'isChildOf' => $child->isChildOf($parent)
    ]);
});

Route::get('/depth-related-to/{categoryId}/{relatedCategoryId}', function ($categoryId, $relatedCategoryId) {
    $category = Category::findOrFail($categoryId);
    $relatedCategory = Category::findOrFail($relatedCategoryId);

    return response()->json([
        'depthRelatedTo' => $category->depthRelatedTo($relatedCategory)
    ]);
});
```

Everything seems to work correctly. Feedback is welcome.

Thanks in advance.

### CODE REVIEW

**Comment by staudenmeir** [comment]
Hi @PaperTurtle,
Thanks, I'll take a look.

**Comment by PaperTurtle** [comment]
Thanks @staudenmeir,
I'm looking forward to your review!

File: src/Eloquent/HasRecursiveRelationships.php [hunk 1]
```diff
@@ -2,11 +2,41 @@
...
 use Staudenmeir\LaravelCte\Eloquent\QueriesExpressions;
 
 trait HasRecursiveRelationships
 {
     use HasAdjacencyList;
     use QueriesExpressions;
+
+    public function isParentOf(Model $model)
+    {
+        if (!$this->relationLoaded('children')) {
```
Comment (staudenmeir) [inline src/Eloquent/HasRecursiveRelationships.php:16]: The `if {}` here (and in the other methods) is not necessary. `$this->children` automatically loads the relationship if necessary.

File: src/Eloquent/HasRecursiveRelationships.php [hunk 2]
```diff
@@ -2,11 +2,41 @@
...
+        return $this->parent ? $this->parent->is($model) : false;
+    }
+
+    public function depthRelatedTo(Model $model)
+    {
+        if (!$this->relationLoaded('ancestors')) {
+            $this->load('ancestors');
+        }
+
+        $index = $this->ancestors->search(function ($ancestor) use ($model) {
```
Comment (staudenmeir) [inline src/Eloquent/HasRecursiveRelationships.php:36]: Please use the `bloodline` relationship so you can also find the related model within the descendants.

**Comment by PaperTurtle** [comment]
@staudenmeir Thanks for the feedback, I've implemented your suggestions.

**Comment by staudenmeir** [comment]
Thanks! I've renamed the third method to `getDepthRelatedTo()` and will release a new version soon.

**Comment by staudenmeir** [comment]
I've released a new version.

**Pull request merged**